webui/react/src/pages/ProjectDetails.tsx

```diff
@@ -883,6 +902,12 @@ const ProjectDetails: React.FC = () => {
             ContextMenu={ContextMenu}
             dataSource={experiments}
             loading={isLoading}
+            numOfPinned={
+              Array.from(intersection(
+                new Set(experiments.map((e) => e.id)),
+                new Set(settings.pinned),
+              )).length
+            }
```

Comment:
question: why can't this just be `settings.pinned.length`?

Reply:
Because of the projects. I store all pinned items in the userSettings array. 
I originally wanted to use key value data type but userSettings doesn't support it. 
That's one of reasons why I wanna recreate userSettings